src/Controls/src/Core/VisualElement.cs

```diff
@@ -1227,9 +1266,19 @@ static void OnRequestChanged(BindableObject bindable, object oldvalue, object ne
 
 		void IPropertyPropagationController.PropagatePropertyChanged(string propertyName)
 		{
+			if (propertyName == null || propertyName == IsEnabledProperty.PropertyName)
+				CommandElement.RefreshPropertyValue(this, IsEnabledProperty, _isEnabledExplicit);
+
 			PropertyPropagationExtensions.PropagatePropertyChanged(propertyName, this, ((IVisualTreeElement)this).GetVisualChildren());
 		}
 
+		/// <summary>
+		/// This method must always be called if some event occurs and the value of
+		/// the IsEnabledCore property will change.
+		/// </summary>
+		internal void RefreshIsEnabledProperty() =>
+			(this as IPropertyPropagationController)?.PropagatePropertyChanged(VisualElement.IsEnabledProperty.PropertyName);
```

Comment:
If I change this line to: 

```
CommandElement.RefreshPropertyValue(this, IsEnabledProperty, _isEnabledExplicit);
```

All the unit tests still pass. 
Does it make sense to just change it. 

The additional code added to `OnIsEnabledPropertyChanged` seems to take care of the propagation. Unless you are intentionally wanting the propagation to happen before the  `OnIsEnabledPropertyChanged` logic is called

Reply:
ah, yeah that could be it... Probably right now there is a double propagation? Let me do that.